### PR TITLE
fix: endorsement modal

### PR DESCRIPTION
**Pull request by pseudobun**

### CODE REVIEW

File: packages/dapp/src/components/EndorseForm/index.tsx [hunk 1]
```diff
@@ -282,8 +258,79 @@ export const EndorseForm = ({ endorsee }: EndorseeProps) => {
...
 
   useEffect(() => {
-    if (isReceiptLoading) setEndorsementModalOpen(true);
-  }, [isReceiptLoading]);
+    if (isReceiptLoaded) setEndorsementModalOpen(true);
+  }, [isReceiptLoaded]);
+
+  useEffect(() => {
+    if (transactionHash === undefined) return;
+    const transactionReceipt = waitForTransactionReceipt(config, {
```
Comment (martines3000) [inline packages/dapp/src/components/EndorseForm/index.tsx:266]: We already use `useWaitForTransactionReceipt` above.

File: packages/dapp/src/components/EndorseForm/index.tsx [hunk 2]
```diff
@@ -107,10 +108,13 @@ export const EndorseForm = ({ endorsee }: EndorseeProps) => {
     config: config,
   });
 
-  const { data: txReceipt, isLoading: isReceiptLoading } =
-    useWaitForTransactionReceipt({
-      hash: transactionHash,
-    });
+  const {
+    data: txReceipt,
+    isSuccess: isReceiptLoaded,
```
Comment (martines3000) [inline packages/dapp/src/components/EndorseForm/index.tsx:113]:
```suggestion
    isSuccess: isReceiptSuccess,
```

File: packages/dapp/src/components/EndorseForm/index.tsx [hunk 3]
```diff
@@ -208,43 +212,15 @@ export const EndorseForm = ({ endorsee }: EndorseeProps) => {
     const _value = parseEther(donationValue ?? '0') + endorsementPrice;
     const _displayValue = `${platform}:${displayValue}`;
 
-    writeContractAsync({
+    await writeContractAsync({
```
Comment (martines3000) [inline packages/dapp/src/components/EndorseForm/index.tsx:215]: We don't need to await, as we use `.then`

File: packages/dapp/src/components/EndorseForm/index.tsx (hunk 1 above)
Comment (andyv09) [reply packages/dapp/src/components/EndorseForm/index.tsx:266]: This one is needed for its promise so toast.promise works. The other one is used for everything else.

File: packages/dapp/src/components/EndorseForm/index.tsx [hunk 4]
```diff
@@ -282,8 +258,79 @@ export const EndorseForm = ({ endorsee }: EndorseeProps) => {
...
+                size="sm"
+                className="p-0 pl-1 text-xs h-4"
+                onClick={() =>
+                  window.open(
+                    `${EXPLORERS[chainId!]}/tx/${transactionHash}`,
+                    '_blank'
+                  )
+                }
+              >
+                {`${transactionHash.slice(0, 4)}...${transactionHash.slice(
```
Comment (martines3000) [inline packages/dapp/src/components/EndorseForm/index.tsx:311]: We can probably use `formatAddress` here.

**Review by martines3000 — APPROVED** [review]

**Pull request merged**